src/lib.rs

```diff
@@ -36,6 +36,90 @@
 //! This crate provides utilities that make writing both clients and servers easier. You'll find
 //! the client helper as `Client` in the protocol module you're working with (e.g.,
 //! [`pipeline::Client`]), and the server helper as `Server` in the same place.
+//!
+//! # Example
+//! ```rust
+//! # type StdError = Box<dyn std::error::Error + Send + Sync + 'static>;
+//! # use std::pin::Pin;
+//! # use std::boxed::Box;
+//! # use tokio::sync::mpsc;
+//! # use tokio::io::{AsyncWrite, AsyncRead};
+//! # use futures_core::task::{Context, Poll};
+//! # use futures_util::{never::Never, future::{poll_fn, ready, Ready}};
+//! # use tokio_tower::pipeline;
+//! /// Wrapper around our transport.
+//! /// Need to implement Sink and Stream for this.
+//! struct Pair {
+//!     rcv: mpsc::Receiver<String>,
+//!     snd: mpsc::Sender<String>,
+//! }
+//!
+//! impl futures_sink::Sink<String> for Pair {
+//!     type Error = StdError;
+//!
+//!     fn poll_ready(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
+//!         self.snd.poll_ready(cx).map_err(|e| e.into())
```

Comment:
```suggestion
//!         Ok(self.snd.poll_ready(cx)?)
```

Reply:
I _think_ that should work?

Reply:
It's `Poll::map_err`, not `Result::map_err`.
So it would have to be
```rust
self.snd.poll_ready(cx).map(|e| Ok(e?))
```